numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -2555,34 +2556,36 @@ object_arrtype_dealloc(PyObject *v)
  * #TYPE = BYTE, SHORT, INT, LONG, LONGLONG, UBYTE, USHORT, UINT, ULONG,
  *         ULONGLONG, HALF, FLOAT, DOUBLE, LONGDOUBLE, CFLOAT, CDOUBLE,
  *         CLONGDOUBLE, STRING, UNICODE, OBJECT#
- * #work = 0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,z,z,0#
- * #default = 0*17,1*2,2#
  */
 
-#define _NPY_UNUSED2_1
-#define _NPY_UNUSED2_z
-#define _NPY_UNUSED2_0 NPY_UNUSED
-#define _NPY_UNUSED1_0
-#define _NPY_UNUSED1_1
-#define _NPY_UNUSED1_2 NPY_UNUSED
+#if @TYPE@_IS_OBJECT
```

Comment:
I suppose these should be `#if _@TYPE@_IS_OBJECT`? (Was going to add `defined` first, but that is unnecessary I guess).
I think I like this style, although did not look through all yet. IIRC one thing that I was wondering if we cannot hardcode the result of `PyTuple_GET_ITEM(cls.tp_bases, num)`

Reply:
Good catch, thanks